core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -708,6 +708,11 @@ private void finishDoStop(CompletableFuture<Void> future) {
 
             CompletableFutures.successfulAsList(completionFutures, cause -> null)
                               .thenRunAsync(() -> future.complete(null), config.startStopExecutor());
+            try {
+                config.dependencyInjector().close();
```

Comment:
How about adding `closeOnStop` for the dependency injector so that a user can choose to manage its life cycle manually?